Services/BackendService.cs

```diff
@@ -51,12 +54,79 @@ public async Task CreateAccountAsync(UserCredentials credentials)
                 await _client.Register(credentials.Username, credentials.Password);
                 Console.WriteLine("Register user success");
             }
-            catch(RestException e)
+            catch (RestException e)
             {
                 Console.WriteLine($"Failed to Register user: {e.Message}");
 
                 throw; //throw the exception
             }
         }
+
+        /// <summary>
+        /// This
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public async Task<List<IItemStack<ICard>>?> GetCardsAsync(ICardQuery query)
+        {
+            //Fetch cards if they are not caches yet
+            try
+            {
+                if (_client.User != null)
+                {
+                    if (query != null) //query must not be null
+                    {
+                        var cardsManager = _client.User.Cards;
+                        Console.WriteLine("Attempting to get user cards");
+                        var cards = await cardsManager.GetCards(query);
+                        return cards; //return cards
+                    }
+                    else
+                    {
+                        throw new Exception("Invalid card filter query");
+                    }
+                }
+                else
+                {
+                    throw new Exception("Invalid user");
+                }
+            }
+            catch (RestException)
+            {
+                throw; // throw the exception
+            }
```

Comment:
if we arent doing additional processing with the exception there is no reason to catch it